tasks.py

```diff
@@ -678,6 +677,73 @@ def get_day_of_week_code(day):
     __schedule_json_to_dynamo(ctx, schedule_data=schedule_data)
 
 
+@task(pre=[production_failsafe])
+def import_fas_schedule_from_csv(ctx, filepath):
+    columns = (
+        "course_code",
+        "title",
+        "day",
+        "start",
+        "end",
+        "zoom_link",
+        "opencast_series_link",
+        "crn",
+        "producer",
+        "sit_in"
+    )
+    day_of_week_map = {
+        "Monday": "M",
+        "Tuesday": "T",
+        "Wednesday": "W",
+        "Thursday": "R",
+        "Friday": "F"
+    }
+    with open(filepath, "r") as f:
+        reader = csv.reader(f)
+        next(reader)
+        rows = [ dict(zip(columns, row)) for row in reader]
+
+    schedule_data = {}
+    for row in rows:
+
+        zoom_link = urlparse(row["zoom_link"])
+        if not zoom_link.scheme.startswith("https"):
+            print("Invalid zoom link value for {}: {}" \
+                  .format(row["course_code"], row["zoom_link"]))
+            continue
+        zoom_series_id = urlparse(row["zoom_link"]).path.split("/")[-1]
+        schedule_data.setdefault(zoom_series_id, {})
+        schedule_data[zoom_series_id]["zoom_series_id"] = zoom_series_id
+
+        opencast_series_id = urlparse(row["opencast_series_link"]) \
+            .fragment.replace("/", "")
+        schedule_data[zoom_series_id]["opencast_series_id"] = opencast_series_id
+
+        topic = "{} - {}".format(row["course_code"], row["title"])
+        schedule_data[zoom_series_id]["opencast_topic"] = topic
+
+        day = day_of_week_map.get(row["day"])
+        if day is None:
+            raise Exit("Got bad day value: {}".format(row["day"]))
+
+        schedule_data[zoom_series_id].setdefault("Days", set())
+        schedule_data[zoom_series_id]["Days"].add(day)
+
+        schedule_data[zoom_series_id].setdefault("Time", set())
+        time_object = datetime.strptime(row["start"], "%H:%M")
+        schedule_data[zoom_series_id]["Time"].update([
+            datetime.strftime(time_object, "%H:%M"),
+            (time_object + timedelta(minutes=30)).strftime("%H:%M"),
+            (time_object + timedelta(hours=1)).strftime("%H:%M")
+        ])
+
+    for id, item in schedule_data.items():
+        item["Days"] = list(item["Days"])
+        item["Time"] = list(item["Time"])
+
+    __schedule_json_to_dynamo(ctx, schedule_data=schedule_data)
```

Comment:
This (__schedule_json_to_dynamo) prints out "schedule updated to" which is kind of useless when the schedule is this long and it hides the error messages printed out by your new code (i.e. "Invalid zoom link value for...")
Perhaps immediately, we should just remove that "schedule updated to" printout. What would be better would be to print out some kind of a diff.

Reply:
yep, diff before updating would be even better. I also would like to look into a task arg that does a dry run and another that creates a backup of the table.